test/main.js

```diff
@@ -164,6 +164,33 @@ test('custom headers', async t => {
 	await server.close();
 });
 
+test('remove custom header by extending instance', async t => {
+	const server = await createTestServer();
+	server.get('/', (request, response) => {
+		response.send(request.headers);
+		response.end();
+	});
+
+	const original = ky.create({
+		headers: {
+			rainbow: 'rainbow',
+			unicorn: 'unicorn'
+		}
```

Comment:
Can you also test what happens when the `original` instance was created with `new Headers()` instead of a plain object? I'm not certain it will work correctly.

Reply:
I added this test, and as you said: it faild. 
What we should get it this case? I`m no sure that we want to mix plain object and `Headers` instance.

```
const original = ky.create({
	headers: new Headers({
		rainbow: 'rainbow',
		unicorn: 'unicorn'
	})
});
const extended = original.extend({
	headers: {
		rainbow: undefined
	}
});
```

Reply:
Well people use `new Headers()`. We have explicit support for it in our API. So I'd at least prefer to it to not be undefined behavior. What should Ky do in that case? Whatever it is, it should have a test.

Reply:
Ok, so I updated expected values in this test